### PR TITLE
Fixes for docker 17.06.0

### PR DESCRIPTION
**Pull request by johnflavin**
In fixing #822, I saw that a lot of tests were failing on docker `17.06.0`. I fixed all the errors in both tests and code so that the tests run cleanly on my machine. 

However, we can't test `17.06.0` on travis yet because it isn't in the [docker apt repo](https://apt.dockerproject.org/repo/).

### CODE REVIEW

**Comment by olegtikhonov** [comment]
Is there a way to get the "fixed" jar somehow? with tag .... "alpha/beta"?

File: src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java [hunk 1]
```diff
@@ -5099,13 +5117,22 @@ public void testCreateServiceWithSecretHostnameHostsAndHealthcheck() throws Exce
             .build();
 
     final String[] commandLine = {"ping", "-c4", "localhost"};
+    final long interval = dockerApiVersionLessThan("1.30") ? 30L : 30000000L;
```
Comment (mattnworb) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: did they change the unit in between api versions? 😬

Comment (johnflavin) [inline src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java]: I think they did, yes. Or maybe they were lenient about the units before, and started being more strict. I think these were always intended to be in nanoseconds (at least [our javadoc](https://github.com/spotify/docker-client/blob/master/src/main/java/com/spotify/docker/client/messages/ContainerConfig.java#L358) says they are in nanoseconds).

**Comment by johnflavin** [comment]
I can add something to the change log, but I want to be careful about the wording. I didn't actually bring us up to full compatibility with the new API. I mostly just made sure the tests we run are asserting accurate stuff, and in a couple places I made small modifications to the code where the tests exposed things that were broken.

**Comment by mattnworb** [comment]
@johnflavin sounds good to me, I think any extra info in the changelog is helpful and better than none. Thanks!

**Pull request merged**